pkg/agent/packetcapture/packetcapture_controller_test.go

```diff
@@ -0,0 +1,569 @@
+
+type fakePacketCaptureController struct {
+	*Controller
+	kubeClient         kubernetes.Interface
+	mockController     *gomock.Controller
+	crdClient          *fakeversioned.Clientset
+	crdInformerFactory crdinformers.SharedInformerFactory
+	informerFactory    informers.SharedInformerFactory
+}
+
+func newFakePacketCaptureController(t *testing.T, runtimeObjects []runtime.Object, initObjects []runtime.Object) *fakePacketCaptureController {
+	controller := gomock.NewController(t)
+	objs := []runtime.Object{
+		&pod1,
+		&pod2,
+		&pod3,
+		&secret1,
+	}
+	objs = append(objs, generateTestSecret())
+	if runtimeObjects != nil {
+		objs = append(objs, runtimeObjects...)
+	}
+	kubeClient := fake.NewSimpleClientset(objs...)
+	crdClient := fakeversioned.NewSimpleClientset(initObjects...)
+	crdInformerFactory := crdinformers.NewSharedInformerFactory(crdClient, 0)
+	packetCaptureInformer := crdInformerFactory.Crd().V1alpha1().PacketCaptures()
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
+
+	ifaceStore := interfacestore.NewInterfaceStore()
+	addPodInterface(ifaceStore, pod1.Namespace, pod1.Name, []string{pod1IPv4, ipv6}, pod1MAC.String(), int32(ofPortPod1))
+	addPodInterface(ifaceStore, pod2.Namespace, pod2.Name, []string{pod2IPv4}, pod2MAC.String(), int32(ofPortPod2))
+
+	pcController, _ := NewPacketCaptureController(
+		kubeClient,
+		crdClient,
+		packetCaptureInformer,
+		ifaceStore,
+	)
+	pcController.sftpUploader = &testUploader{}
+	pcController.captureInterface = &testCapture{}
+	t.Setenv("POD_NAME", "antrea-agent")
+	return &fakePacketCaptureController{
+		Controller:         pcController,
+		kubeClient:         kubeClient,
+		mockController:     controller,
+		crdClient:          crdClient,
+		crdInformerFactory: crdInformerFactory,
+		informerFactory:    informerFactory,
+	}
+}
+
+func addPodInterface(ifaceStore interfacestore.InterfaceStore, podNamespace, podName string, podIPs []string, podMac string, ofPort int32) {
+	containerName := k8s.NamespacedName(podNamespace, podName)
+	var ifIPs []net.IP
+	for _, ip := range podIPs {
+		ifIPs = append(ifIPs, net.ParseIP(ip))
+	}
+	mac, _ := net.ParseMAC(podMac)
+	ifaceStore.AddInterface(&interfacestore.InterfaceConfig{
+		IPs:                      ifIPs,
+		MAC:                      mac,
+		InterfaceName:            util.GenerateContainerInterfaceName(podName, podNamespace, containerName),
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{PodName: podName, PodNamespace: podNamespace, ContainerID: containerName},
+		OVSPortConfig:            &interfacestore.OVSPortConfig{OFPort: ofPort},
+	})
+}
+
+// TestPacketCaptureControllerRun was used to validate the whole run process is working. It doesn't wait for
+// the testing pc to finish. on sandbox env, no good solution to open raw socket.
+func TestPacketCaptureControllerRun(t *testing.T) {
+	// create test os
+	defaultFS = afero.NewMemMapFs()
+	defaultFS.MkdirAll("/tmp/antrea/packetcapture/packets", 0755)
```

Comment:
this doesn't look like it is used (anymore?)

Reply:
it's used but it's not checked. The capture process involves create files in `/tmp`, so usually it's set to mem os.